src/main/java/org/folio/edge/orders/OrdersHandler.java

```diff
@@ -99,107 +98,98 @@ protected void handleResponse(RoutingContext ctx, HttpClientResponse resp) {
       body.append(buf);
     })
         .endHandler(v -> {
-          ctx.response()
-              .setStatusCode(resp.statusCode());
+          ctx.response().setStatusCode(resp.statusCode());
 
           if (body.length() > 0) {
             String respBody = body.toString();
-
-            if (logger.isDebugEnabled()) {
-              logger.debug("status: {} response: {}", + resp.statusCode(), respBody);
-            }
-
             try {
-              String xml = StringUtils.EMPTY;
-              if (respBody.matches(VALIDATE_SUCCESS)) {
-                xml = respBody;
+              if (respBody.matches(VALIDATE_GOBI_XML_SUCCESS)) {
```

Comment:
Why do we need to do any of this?  Can't we just send back the response as-is?  The point of this feature is to avoid having to make code changes when adding new vendor integrations (i.e. types).  If we continue to perform response validation and/or parse the response that becomes difficult.  It's probably better to just trust that the vendor-specific backend module is returning a valid/well-formed response and let the edge API be "dumb"... just pass the response along.

Reply:
Also note that in the case of error responses, I don't think we want to try to map things to a common error schema, regardless of the content type....  Different vendor systems will undoubtedly have differing expectations about what an error response looks like.  So, just like the non-error responses, I think we want to keep this generic and just pass the response from the backend to the client as-is.  

The one exception is when something fails in the edge module itself e.g. invalid API key, or inability to get credentials/logic, etc.  In these cases I think we probably need to allow for some error templates to be supplied via configuration and do some substitution, e.g. an XML snippet can be provided for "invalid API key" responses.  The snippet could have placeholders for things like {message}, {code}, etc.   This however sounds like another story.

**In the short term, I would  just continue to use the error responses you have defined here for these types of errors.**

To summarize, here's my take on what should happen:
* For errors in the edge API, use the error responses you've defined here (for now)
* Create a JIRA for implementing error templates, specified via configuration
* Return **all** responses from the backend as-is - success, failure, etc.
* Responses from the backend should not be parsed or validated
* The RAML cannot specify schemas for any responses since they will differ depending on the 'type' query argument.

Reply:
OK, after a discussion and closer code review it seems that the proposed plan above needs to be tweaked because mod-gobi currently returns text/plain.

1. If the backend returns text/plain, wrap in the XML/JSON error response format
2. Story for mod-gobi to return error XML conforming to the schema currently in use - currently returns text/plain
3. Story for edge-orders to remove the wrapping and return responses as-is (depends on the mod-gobi work above)
4. Models and scheams for gobi order responses (success) can be removed - no longer needed
5. For now, leave the examples in RAML - remove as part of #3